scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -333,9 +333,9 @@ object Ctor {
 @ast class Template(early: List[Stat],
                     inits: List[Init],
                     self: Term.Param,
-                    stats: Option[List[Stat]]) extends Tree {
+                    stats: List[Stat]) extends Tree {
```

Comment:
On one hand, I am excited to see this complication gone. With Init and Term.NewAnonymous (Term.NewRefinement is better btw!), it is no longer necessary to distinguish between `object a {}` and `object a`. cc/ @DavidDudson